kolibri/core/assets/src/views/InteractionList/__tests__/index.spec.js

```diff
@@ -0,0 +1,51 @@
+import { render, fireEvent, screen } from '@testing-library/vue';
+import VueRouter from 'vue-router';
+import InteractionList from '../';
+
+const renderComponent = props => {
+  const defaultProps = {
+    interactions: [{ type: 'hint' }, { type: 'answer', correct: true }, { type: 'error' }],
+    selectedInteractionIndex: 0,
+  };
+
+  return render(InteractionList, {
+    props: {
+      ...defaultProps,
+      ...props,
+    },
+    routes: new VueRouter(),
+  });
+};
+
+describe('InteractionList', () => {
+  // Check if the selected interaction has the correct cursor
+  // Auto cursor means that the interaction is selected, and pointer cursor means that it is not
+  const isSelected = iteractionItem => iteractionItem.style.cursor === 'auto';
+
+  test('renders correctly and the selected iteraction changes correctly on user clicks', async () => {
```

Comment:
Here we are testing several things, we can break up this test, and give a more meaningful name.

1. renders corectly all interactions. (We specifically set the 3 interactions and check the length is correct.
2. just the selectedInteractionIndex is actually selected. (Here we specifically set the `selectedInteractionIndex` prop).
3. renders attempt message correctly.
4. Doesnt emit the selected event when the user clicks on the interaction that is already selected.
5. Emits the selected event when the user clicks on a new interaction item.

Reply:
Got it. I feel that that the points `2`  and `3` can be clubbed into a single test, and we can wrap all of them in a `describe` block to provide more semantic meaning. I'll update the test suite and request a review when done. Thanks for your time!